native/cocos/renderer/gfx-vulkan/VKPipelineCache.cpp

```diff
@@ -0,0 +1,120 @@
+/****************************************************************************
+ Copyright (c) 2019-2023 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#include "VKPipelineCache.h"
+#include "gfx-base/GFXUtil.h"
+#include "base/BinaryArchive.h"
+#include <fstream>
+
+namespace cc::gfx {
+
+static const char* fileName = "/pipeline_cache_vk.bin";
+
+static const uint32_t MAGIC   = 0x4343564B; // "CCVK"
+static const uint32_t VERSION = 1;
+
+static void loadData(ccstd::vector<char> &data) {
+    auto path = getPipelineCacheFolder() + fileName;
+    std::ifstream stream(path, std::ios::binary);
+    if (!stream.is_open()) {
+        CC_LOG_INFO("Load program cache, no cached files.");
+        return;
+    }
+
+    uint32_t magic = 0;
+    uint32_t version = 0;
+
+    BinaryInputArchive archive(stream);
+    auto loadResult = archive.load(magic);
+    loadResult &= archive.load(version);
+
+    uint32_t size = 0;
+    if (loadResult && magic == MAGIC && version >= VERSION) {
+        loadResult &= archive.load(size);
+        data.resize(size);
+        loadResult &= archive.load(data.data(), size);
+    }
+    if (loadResult) {
+        CC_LOG_INFO("Load pipeline cache success.");
+    }
+}
```

Comment:
It is better to implement them in anonymous namespace.

Reply:
Both static and anonymous namespace allow object with internal linkage. I used to use `static`. `namespace` should be better as it works for more things. I have replaced with namespace.